### PR TITLE
133 uncertainty visualization

### PR DESCRIPTION
**Pull request by lillijo**
current status, stable but a bit slow

### CODE REVIEW

File: src/components/ProjectDetailsPanel/project-details-panel.module.css [hunk 1]
```diff
@@ -27,11 +27,40 @@
...
+
 .abstractText{
     color: white;
-}
\ No newline at end of file
+    width: 100%;
+    clear: both;
+    overflow-y: scroll;
+    height: 300px;
+}
```
Comment (mx-e) [inline src/components/ProjectDetailsPanel/project-details-panel.module.css]: this works, but a more modern approach would be using flex box and/or grid 

https://css-tricks.com/snippets/css/complete-guide-grid/
https://css-tricks.com/snippets/css/a-guide-to-flexbox/

not really necessary for such simple cases as this, but good to know

File: src/components/TimeLine/time-line-view.module.css [hunk 2]
```diff
@@ -48,6 +48,21 @@
...
+  overflow: auto;
+}
+
+.ktasPlot > .ktasLine{
+  font-family: IBM_Plex_Mono;
+  overflow: auto;
+  height: auto !important;
+  width: auto !important;
+  font-size: 10px;
+}
```
[the remark this answers is not in the document]
Comment (mx-e) [reply src/components/TimeLine/time-line-view.module.css]: if i disable both commands in my dev-tools, nothing happens

[the remark this answers is not in the document]
Comment (mx-e) [reply src/components/TimeLine/time-line-view.module.css]: same goes for overflow both in ktasPlot and ktasLine

[the remark this answers is not in the document]
Comment (mx-e) [reply src/components/TimeLine/time-line-view.module.css]: font family is inherited, so if you set it in ktasPlot you don't have to set it in ktasLine and vice versa (you can try such things in the Chrome dev tools under inspect and check the boxes of certain css commands to see what they do)

File: src/components/TimeLine/time-line.js [hunk 3]
```diff
@@ -117,4 +122,51 @@ const processData = (data, colors) => {
...
+  if (!ktas || ktas === []) return [];
+
+  let ktasSplitYears = [];
+  for (let ktaKey in ktas) {
+    let startDate = parseInt(ktas[ktaKey].start_date);
+    let endDate = parseInt(ktas[ktaKey].end_date);
+    if (isNaN(startDate) || startDate === "") startDate = 1900;
+    if (isNaN(endDate) || endDate === "") endDate = startDate;
+    if(startDate > 2000){
+      for (let targ in ktas[ktaKey].targetgroups){
```
Comment (mx-e) [inline src/components/TimeLine/time-line.js]: there's really no reason for using confusing short forms, IDEs will auto-complete anyway and its much more readable

File: src/components/TimeLine/time-line.js [hunk 4]
```diff
@@ -117,4 +122,51 @@ const processData = (data, colors) => {
   return dataSplitYears;
 };
 
+const processKtas = (ktas) => {
+
+  if (!ktas || ktas === []) return [];
+
+  let ktasSplitYears = [];
```
Comment (mx-e) [inline src/components/TimeLine/time-line.js]: ktasSplitYears ? 

is it a list of ktas by year ? 

ktasByYear or ktasYearlyBuckets  (or something better, these are pretty terrible as well :D)

File: src/components/TimeLine/time-line-view.js [hunk 5]
```diff
@@ -216,13 +219,35 @@ class TimeLineView extends Component {
...
-      <div style={{ marginTop: (this.props.height * 0.75) / 2 }}>
+      <div style={{ marginTop: (this.props.height * 0.05) }}>
+        <div className={styles.ktasPlot}>
+          {Object.keys(this.state.ktasSplitYears).map(targetgroup => {
+              if(this.state.ktasSplitYears[targetgroup].length <3){
+                return "";
+              }
+            return(<div key={targetgroup} className={styles.ktasLine}>
+              <span  >{targetgroup}</span> <br/>
+              <svg height={40} width={this.state.width} >
```
Comment (mx-e) [inline src/components/TimeLine/time-line-view.js]: make a named constant

Comment (lillijo) [reply src/components/TimeLine/time-line-view.js]: what do you mean exactly?

File: src/components/ClusterMap/cluster-map-view.js [hunk 6]
```diff
@@ -92,6 +98,23 @@ export default class ClusterMapView extends React.Component {
     );
   }
 
+  wrapText = (text) => {
```
[the remark this answers is not in the document]
Comment (lillijo) [reply src/components/ClusterMap/cluster-map-view.js]: sadly \n does not work in svgs. i tried a lot. maybe didnt try hard enough?

File: src/util/utility.js [hunk 7]
```diff
@@ -129,6 +148,16 @@ export const getFieldColor = field => {
     : "#989aa1"; // default color field
 };
 
+export const getFieldIcon = field => {
+  return fieldsMapping.find(e => e.name === field)
+    ? iconPaths[fieldsMapping.find(e => e.name === field).field % iconPaths.length] : field==="pfeil" ? IconTimelinePaths[1] : ""
```
[the remark this answers is not in the document]
Comment (lillijo) [reply src/util/utility.js]: still in work, but will think of your suggestions when its icon time! ^^

File: src/components/ClusterMap/cluster-map-view.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (mx-e) [reply src/components/ClusterMap/cluster-map-view.js]: maybe `<br>` ?

File: src/store/reducer/reducer.js [hunk 8]
```diff
@@ -206,8 +204,14 @@ const updateProjectsData = (state, action) => {
       new Date(project.funding_start_year).getFullYear(),
       new Date(project.funding_end_year).getFullYear()
     ];
-    project.collections = typeof project.sammlungen != "undefined" && project.sammlungen[0] != null ? project.sammlungen : ["Keine Sammlung"];
-    project.infrastructure = typeof project.infrastruktur != "undefined" && project.infrastruktur[0] != null ? project.infrastruktur : ["Kein Laborgerät"];
+    project.collections =
+      project.sammlungen && project.sammlungen[0] != null
```
Comment (mx-e) [inline src/store/reducer/reducer.js]: project.sammlungen[0] does the same to my knowledge

File: src/components/TimeLine/TargetgroupRow.js [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+            x2={this.props.width}
+            y2={this.props.height}
+            stroke="#717071"
+            fill="none"
+          />
+        </svg>
+      </div>
+    );
+  }
+}
```
Comment (mx-e) [inline src/components/TimeLine/TargetgroupRow.js]: If you make a component completely without state and connection to redux, make it stateless, i.e.: 

`const componentName = props => {`
`return (`
`<HTML> `
`)`
`}`

or even

`const componentName = props => (`
`<HTML> `
`)`

you can access props with `props.x`

File: src/store/reducer/reducer.js (hunk 8 above)
Comment (lillijo) [reply src/store/reducer/reducer.js]: oh ja, stupid me ^^

**Pull request merged**